File: frontend/src/types/agent.ts
export type AgentStatus = "Active" | "Idle" | "Error"

export interface ToolCall {
  id: string
  tool: string
  input: string
  output: string
  durationMs: number
  timestamp: string
}

export interface Agent {
  id: string
  name: string
  role: string
  order: number
  purpose: string
  input: string
  output: string
  status: AgentStatus
  memory: string[]
  skills: string[]
  toolCalls: ToolCall[]
  executionTimeMs: number
  confidence: number
  lastRun: string
  countries: string[]
}

export interface SupervisorFlowEdge {
  id: string
  source: string
  target: string
  animated: boolean
  label: string
}

File: backend/app/schemas/agent.py
from typing import Literal
from pydantic import BaseModel

AgentStatus = Literal["Active", "Idle", "Error"]


class ToolCall(BaseModel):
    id: str
    tool: str
    input: str
    output: str
    durationMs: int
    timestamp: str


class Agent(BaseModel):
    id: str
    name: str
    role: str
    order: int
    purpose: str
    input: str
    output: str
    status: AgentStatus
    memory: list[str]
    skills: list[str]
    toolCalls: list[ToolCall]
    executionTimeMs: int
    confidence: float
    lastRun: str
    countries: list[str]


class SupervisorFlowEdge(BaseModel):
    id: str
    source: str
    target: str
    animated: bool
    label: str
